capstone-frontend/src/components/modal.jsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState, useContext, useRef, createContext } from "react";
+import ApplicationContext from "../applicationContext";
+import Modal from "react-modal";
+
+const customStyles = {
+    content: {
+      top: "50%",
+      left: "50%",
+      marginRight: "-50%",
+      transform: "translate(-50%, -50%)",
+    },
+  };
+
+function ModalPopulate() {
+    const { modalIsOpen, setIsOpen, socket } = useContext(ApplicationContext);
+    const websocket = socket.current;
```

Comment:
I don't think it's correct to read this out here, you should only access `socket.current` in the effect. Because it may have changed between when this code runs and when the effect runs

Reply:
In other words, move this line inside of the useEffect